File: Problem 2 - Add Two Numbers/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    
    var carry = 0;
    var result = new ListNode();
    var current = result;
    var current_sum = 0;
    
    while(l1 || l2 || carry > 0) {
        current_sum = carry + (l1 ? l1.val : 0) + (l2 ? l2.val : 0);
        carry = (current_sum > 9) ? 1 : 0;
        current.next = new ListNode(current_sum % 10);
        current = current.next;
        l1 = ( l1 ? l1.next : null);
        l2 = ( l2 ? l2.next : null);
    }
    
    return result.next;
    
    
};

File: Problem 2 - Add Two Numbers/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        
        ListNode* result = new ListNode(0);
        int carry = 0;
        
        ListNode* current = result;
        int current_sum = 0;
        
        while(l1 || l2 || carry > 0) {
            current_sum = (l1 ? l1->val : 0) + (l2 ? l2->val : 0) + carry;
            current->next = new ListNode(current_sum % 10);
            current = current->next;
            carry = current_sum/10;
            l1 = (l1 ? l1->next : nullptr);
            l2 = (l2 ? l2->next : nullptr);            
        }
        
        return result->next;
    }
};